src/store/modules/order/actions.ts

```diff
@@ -0,0 +1,184 @@
+    }
+
+    return resp;
+  },
+
+  async deliverShipment ({ dispatch }, order) {
+    emitter.emit("presentLoader");
+
+    const params = {
+      shipmentId: order.shipmentId,
+      statusId: 'SHIPMENT_SHIPPED'
+    }
+
+    let resp;
+
+    try {
+      resp = await OrderService.updateShipment(params)
+      if (resp.status === 200 && !hasError(resp)) {
+        showToast(translate('Order delivered to', {customerName: order.customerName}))
+      } else {
+        showToast(translate("Something went wrong"))
+      }
+    } catch(err) {
+      console.log(err)
+      showToast(translate("Something went wrong"))
+    }
+
+    emitter.emit("dismissLoader")
+    return resp;
+  },
+
+  async packDeliveryItems ({ commit }, shipmentId) {
+    const params = {
+      shipmentId: shipmentId,
+      statusId: 'SHIPMENT_PACKED'
+    }
+    return await OrderService.updateShipment(params)
+  },
+
+  async quickShipEntireShipGroup ({ dispatch }, payload) {
+    emitter.emit("presentLoader")
+
+    const params = {
+      orderId: payload.order.orderId,
+      setPackedOnly: 'Y',
+      dimensionUomId: 'WT_kg',
+      shipmentBoxTypeId: 'YOURPACKNG',
+      weight: '1',
+      weightUomId: 'WT_kg',
+      facilityId: payload.facilityId,
+      shipGroupSeqId: payload.shipGroup
+    }
+    
+    let resp;
+
+    try {
+      resp = await OrderService.quickShipEntireShipGroup(params)
+      if (resp.status === 200 && !hasError(resp) && resp.data._EVENT_MESSAGE_) {
+        /* To display the button label as per the shipmentMethodTypeId, this will only used on orders segment.
+          Because we get the shipmentMethodTypeId on items level in wms-orders API.
+          As we already get shipmentMethodTypeId on order level in readytoshiporders API hence we will not use this method on packed orders segment.
+        */
+        const shipmentMethod = payload.order.items.find((ele: any) => ele.shipGroupSeqId == payload.shipGroup).shipmentMethodTypeId
+        if (shipmentMethod !== 'STOREPICKUP') {
+          const shipmentId = resp.data._EVENT_MESSAGE_.match(/\d+/g)[0]
+          await dispatch('packDeliveryItems', shipmentId).then((data) => {
+            if (!hasError(data) && !data.data._EVENT_MESSAGE_) showToast(translate("Something went wrong"))
+          })
+        }
+        showToast(translate("Order packed and ready for delivery"))
+      } else {
+        showToast(translate("Something went wrong"))
+      }
+    } catch(err) {
+      console.log(err)
+      showToast(translate("Something went wrong"))
+    }
+
+    emitter.emit("dismissLoader")
+    return resp;
+  },
+
+  // TODO: handle the unfillable items count
+  async setUnfillableOrderOrItem ({ dispatch }, order) {
+    emitter.emit("presentLoader");
+    return await dispatch("rejectOrderItems", order).then((resp) => {
+      const refreshPickupOrders = resp.find((response: any) => !(response.data._ERROR_MESSAGE_ || response.data._ERROR_MESSAGE_LIST_))
```

Comment:
We are checking the first success response and showing toast with all items canceled, what if one of the API has error.

Reply:
Sir, we are using `Promise.all` to make the api calls, which will be resolved if all promises are resolved or will immediately be rejected if any of the API rejects.